pgserver/connection_data.go

```diff
@@ -58,11 +58,26 @@ type ConvertedStatement struct {
 	Tag                string
 	PgParsable         bool
 	HasSentRowDesc     bool
+	IsExtendedQuery    bool
 	SubscriptionConfig *SubscriptionConfig
 	BackupConfig       *BackupConfig
 	RestoreConfig      *RestoreConfig
 }
 
+func (cs ConvertedStatement) WithString(s string) ConvertedStatement {
+	return ConvertedStatement{
```

Comment:
Maybe we should rename this field to `QueryString` (and this method to WithQueryString) to improve its clarity.